tpm2/tpm2.go

```diff
@@ -347,15 +349,28 @@ func decodeCreatePrimary(in []byte) (tpmutil.Handle, *rsa.PublicKey, error) {
 	if err != nil {
 		return 0, nil, err
 	}
-	// Endianness of big.Int.Bytes/SetBytes and modulus in the TPM is the same
-	// (big-endian).
-	pubKey := &rsa.PublicKey{N: pub.RSAParameters.Modulus, E: int(pub.RSAParameters.Exponent)}
+	var pubKey crypto.PublicKey
+	switch pub.Type {
+	case AlgRSA:
+		// Endianness of big.Int.Bytes/SetBytes and modulus in the TPM is the same
+		// (big-endian).
+		pubKey = &rsa.PublicKey{N: pub.RSAParameters.Modulus, E: int(pub.RSAParameters.Exponent)}
+	case AlgECC:
+		pubKey = &ecdsa.PublicKey{X: pub.ECCParameters.Point.X, Y: pub.ECCParameters.Point.Y}
+		curve, ok := toGoCurve[pub.ECCParameters.CurveID]
+		if !ok {
+			return 0, nil, fmt.Errorf("can't map TPM EC curve ID 0x%x to Go elliptic.Curve value", pub.ECCParameters.CurveID)
+		}
+		pubKey.(*ecdsa.PublicKey).Curve = curve
```

Comment:
[optional]: Instead of having to cast, why not try to convert the curve value first? Then the Curve field can be filled in directly in the assignment to pubKey:

```go
		curve, ok := toGoCurve[pub.ECCParameters.CurveID]
		if !ok {
			return 0, nil, fmt.Errorf("can't map TPM EC curve ID 0x%x to Go elliptic.Curve value", pub.ECCParameters.CurveID)
		}
		pubKey = &ecdsa.PublicKey{
                      X: pub.ECCParameters.Point.X,
                      Y: pub.ECCParameters.Point.Y,
                      Curve: curve,
                }
```

Reply:
Good call, done